src/api/viewportFeatures.js

```diff
@@ -0,0 +1,81 @@
+// import {pointToTile} from '@mapbox/tilebelt';
+import bboxPolygon from '@turf/bbox-polygon';
+import intersects from '@turf/boolean-intersects';
+
+// function getTilesInViewport(viewport, zoom) {
+
+//   const [minX, minY, maxX, maxY] = viewport;
+//   const tiles = [];
+//   const tileMin = pointToTile(minX, minY, zoom);
+//   const tileMax = pointToTile(maxX, maxY, zoom);
+
+//   for (let x=tileMin[0]; x<=tileMax[0]; x++){
+//     for (let y=tileMax[1]; y<=tileMin[1]; y++) {
+//       tiles.push({x, y});
+//     }
+//   }
+
+//   return tiles;
+// }
+
+// Clip the viewport with the tile and transform to tile coordinates [0..1]
+function prepareViewport(tile, viewport) {
+  // Clip viewport with the tile
```

Comment:
Passing just `bbox` instead of `tile.bbox` and destructuring?
```javascript
function prepareViewport(tile, viewport) {
  const {west, south, east, north} = bbox;
}
```